### PR TITLE
feat: adjust rules to accommodate for Next JS [CORE-3724]

### PR DESCRIPTION
**Pull request by ddrakeatside**

### CODE REVIEW

File: packages/eslint-config-react/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@side/eslint-config-react",
-  "version": "0.8.5",
+  "version": "0.9.0",
```
Comment (hershmire) [inline packages/eslint-config-react/package.json:3]: You don't manage the version. The CI will take care of this automagically based on the commit message that is merged into master.

File: packages/eslint-config-react/rules/react.js [hunk 2]
```diff
@@ -17,5 +17,9 @@ module.exports = {
         ignoreFunctionalComponents: true,
       },
     ],
+    // Next.JS does not require JSX to be in scope
+    'react/react-in-jsx-scope': 'off',
+    // We're okay with this but use it sparingly and only when this is intentional
+    'react/jsx-props-no-spreading': 'off',
```
Comment (hershmire) [inline packages/eslint-config-react/rules/react.js:23]: I did notice the `explicitSpread` option that is set by default: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md#explicitspread. However, you probably just want to allow spreading in general. If so, then this is good.

**Review by hershmire — CHANGES_REQUESTED** [review]

File: packages/eslint-config-react/rules/react.js (hunk 2 above)
Comment (ddrakeatside) [reply packages/eslint-config-react/rules/react.js:23]: It is intentional. If we're going to wrap Material-UI components, the "dump of props" to the wrapped component is usually a good way to keep your wrapper usable.

Comment (ddrakeatside) [reply packages/eslint-config-react/rules/react.js:23]: It's also nice to be able to do something like:

```
entities.map(entity => <EntityItem {...entity} />)
```

Comment (hershmire) [reply packages/eslint-config-react/rules/react.js:23]: Not opposed, just making you aware of that other option before we push the button :).

**Review by hershmire — APPROVED** [review]

**Pull request merged**